node/node_test.go

```diff
@@ -3447,6 +3447,121 @@ func TestNode_GetAccountAccountExistsShouldReturn(t *testing.T) {
 	require.Equal(t, testscommon.TestAddressAlice, recovAccnt.OwnerAddress)
 }
 
+func TestNode_GetAccountAccountWithKeysErrorShouldErr(t *testing.T) {
+	accnt := createAcc(testscommon.TestPubKeyBob)
+	_ = accnt.AddToBalance(big.NewInt(1))
+	expectedErr := errors.New("expected error")
+	accnt.SetDataTrie(
+		&trieMock.TrieStub{
+			GetAllLeavesOnChannelCalled: func(leavesChannels *common.TrieIteratorChannels, ctx context.Context, rootHash []byte, _ common.KeyBuilder, tlp common.TrieLeafParser) error {
+				return expectedErr
+			},
+			RootCalled: func() ([]byte, error) {
+				return nil, nil
+			},
+		})
+
+	accDB := &stateMock.AccountsStub{
+		GetAccountWithBlockInfoCalled: func(address []byte, options common.RootHashHolder) (vmcommon.AccountHandler, common.BlockInfo, error) {
+			return accnt, nil, nil
+		},
+		RecreateTrieCalled: func(rootHash []byte) error {
+			return nil
+		},
+	}
+
+	coreComponents := getDefaultCoreComponents()
```

Comment:
This code part is duplicated with the one from `TestNode_GetAccountAccountWithKeysShouldReturn`. 
Perhaps we can extract it to a common func

Reply:
fixed